crates/time/src/lib.rs

```diff
@@ -0,0 +1,319 @@
+            // exclusive.
+            if era.end.epoch > epoch {
+                let epochs_elapsed = epoch - era.start.epoch;
+                let offset = era.start.slot;
+                let start = offset + era.params.epoch_size * epochs_elapsed;
+                let end = offset + era.params.epoch_size * (epochs_elapsed + 1);
+                return Ok(EpochBounds {
+                    start: start,
+                    end: end,
+                })
+            }
+        }
+        return Err(TimeHorizonError::PastTimeHorizon);
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use hex;
+
+    #[test]
+    fn test_slot_to_time() {
+        let params = EraParams::new(86400, 1000).unwrap();
+        let eras = EraHistory {
+            eras: vec![
+                Summary {
+                    start: Bound {
+                        time: 0,
+                        slot: 0,
+                        epoch: 0,
+                    },
+                    end: Bound {
+                        time: 86400000,
+                        slot: 86400,
+                        epoch: 1,
+                    },
+                    params: params.clone(),
+                },
+                Summary {
+                    start: Bound {
+                        time: 86400000,
+                        slot: 86400,
+                        epoch: 1,
+                    },
+                    end: Bound {
+                        time: 172800000,
+                        slot: 172800,
+                        epoch: 2,
+                    },
+                    params: params.clone(),
+                },
+            ],
+        };
+        let t0 = eras.slot_to_relative_time(172801);
+        assert_eq!(t0, Err(TimeHorizonError::PastTimeHorizon));
+        let t1 = eras.slot_to_relative_time(172800);
+        assert_eq!(t1, Ok(172800000));
+    }
+
+    #[test]
+    fn test_epoch_bounds() {
+        let params = EraParams::new(86400, 1000).unwrap();
+        let eras = EraHistory {
+            eras: vec![
+                Summary {
+                    start: Bound {
+                        time: 0,
+                        slot: 0,
+                        epoch: 0,
+                    },
+                    end: Bound {
+                        time: 864000000,
+                        slot: 864000,
+                        epoch: 10,
+                    },
+                    params: params,
+                },
+            ],
+        };
+        assert_eq!(eras.epoch_bounds(1).unwrap().start, 86400);
+        assert_eq!(eras.epoch_bounds(1).unwrap().end, 172800);
+        assert_eq!(eras.epoch_bounds(10), Err(TimeHorizonError::PastTimeHorizon));
+    }
+
+    #[test]
+    fn test_slot_to_epoch() {
+        let params = EraParams::new(86400, 1000).unwrap();
+        let eras = EraHistory {
+            eras: vec![
+                Summary {
```

Comment:
You could remove duplication by lifting this definition to be a constant within the test module perhaps?